app/models/product.rb

```diff
@@ -1,17 +1,32 @@
 class Product < ApplicationRecord
-  validates :title, :description, :image_url, presence: { message: "can't be blank"}
+
+  validates_with PriceValidator
+  validates :url, presence: true, url: true
+  validates :title, :description, presence: true
   validates :title, uniqueness: true
-  validates :image_url, allow_blank: true, format: {with: %r{\.(gif|jpg|png)\z}i, message: 'must be a URL for GIF, JPG or PNG image.'}
-  validates :price, numericality: { greater_than_or_equal_to: 0.01 }
+  validates :image_url, allow_blank: true, format: { with: IMAGE_URL_REGEX, message: 'must be a URL for GIF, JPG or PNG image.'}
+  validates :permalink, uniqueness: true, format: { with: PERMALINK_REGEX, message: 'is not in valid format'}
+  validates :price, numericality: { greater_than_or_equal_to: 0.01 },if: :price
```

Comment:
add space after comma.
find another way to do so

Reply:
done